File: output/sonarqube_analysis/cursor_solutions/python/results/q17-longest-substring-without-repeating-characters.py
# https://leetcode.com/problems/longest-substring-without-repeating-characters/
# Problem: Longest Substring Without Repeating Characters

#Problem Description:
#Given a string s, find the length of the longest substring without duplicate characters.

# Constraints:
# 0 <= s.length <= 5 * 10^4
# s consists of English letters, digits, symbols and spaces.

#Code Structure:

class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        char_map = {}
        left = 0
        max_length = 0
        
        for right in range(len(s)):
            char = s[right]
            
            # If character is already in the window, move left pointer
            if char in char_map and char_map[char] >= left:
                left = char_map[char] + 1
            
            # Update character's last seen position
            char_map[char] = right
            
            # Update maximum length
            max_length = max(max_length, right - left + 1)
        
        return max_length 
    

File: output/sonarqube_analysis/cursor_solutions/php/results/q17-longest-substring-without-repeating-characters.php
<?php
/** https://leetcode.com/problems/longest-substring-without-repeating-characters */
// Problem: Longest Substring Without Repeating Characters

//Problem Description:
//Given a string s, find the length of the longest substring without duplicate characters.

// Constraints:
// 0 <= s.length <= 5 * 10^4
// s consists of English letters, digits, symbols and spaces.

//Code Structure:

class Solution {

    /**
     * @param String $s
     * @return Integer
     */
    function lengthOfLongestSubstring($s) {
        $charMap = [];
        $left = 0;
        $maxLength = 0;
        
        for ($right = 0; $right < strlen($s); $right++) {
            $char = $s[$right];
            
            // If character is already in the window, move left pointer
            if (isset($charMap[$char]) && $charMap[$char] >= $left) {
                $left = $charMap[$char] + 1;
            }
            
            // Update character's last seen position
            $charMap[$char] = $right;
            
            // Update maximum length
            $maxLength = max($maxLength, $right - $left + 1);
        }
        
        return $maxLength;
    }
}
    